src/utils/thread_helper.cpp

```diff
@@ -944,6 +944,43 @@ void ThreadHelper::DetachGracefullyCallback(void)
 
 #if OTBR_ENABLE_TELEMETRY_DATA_API
 #if OTBR_ENABLE_BORDER_ROUTING
+void ThreadHelper::RetrieveInfraLinkInfo(threadnetwork::TelemetryData::InfraLinkInfo *aInfraLinkInfo)
+{
+    {
+        otSysInfraNetIfAddressCounters addressCounters;
+        uint32_t                       ifrFlags = otSysGetInfraNetifFlags();
+
+        otSysCountInfraNetifAddresses(&addressCounters);
+
+        aInfraLinkInfo->set_name(otSysGetInfraNetifName());
+        aInfraLinkInfo->set_is_up((ifrFlags & IFF_UP) != 0);
+        aInfraLinkInfo->set_is_running((ifrFlags & IFF_RUNNING) != 0);
+        aInfraLinkInfo->set_is_multicast((ifrFlags & IFF_MULTICAST) != 0);
+        aInfraLinkInfo->set_link_local_address_count(addressCounters.mLinkLocalAddresses);
+        aInfraLinkInfo->set_unique_local_address_count(addressCounters.mUniqueLocalAddresses);
+        aInfraLinkInfo->set_global_unicast_address_count(addressCounters.mGlobalUnicastAddresses);
+    }
+
+    //---- peer_br_count
+    {
+        uint8_t                            count = 0;
```

Comment:
How about using `uint16_t` or `uint32_t` for this to be safe (it's unlikely to have more than 255 BRs, but it is technically possible).

What is the type for this in the `.proto` definition? I believe it's defined as `uint32_t` (so maybe using the same type here for consistency?)

Reply:
Yes, it's uint32 in `.proto`. It sounds reasonable to make it consistent!